common/src/main/scala/akka/contrib/persistence/mongodb/MongoReadJournal.scala

```diff
@@ -72,9 +73,12 @@ class ScalaDslMongoReadJournal(impl: MongoPersistenceReadJournallingApi)(implici
 
   def allEvents(offset: Long): Source[EventEnvelope, NotUsed] = {
     val pastSource = impl.currentAllEvents(offset)
-    val realtimeSource = Source.actorRef(100, OverflowStrategy.dropTail).mapMaterializedValue(impl.subscribeJournalEvents)
+    val realtimeSource = Source.actorRef[Event](100, OverflowStrategy.dropTail).mapMaterializedValue(ref => {
+                                                                                  impl.subscribeJournalEvents(ref)
+                                                                                  NotUsed
+                                                                                })
     (pastSource ++ realtimeSource)
-      .filter(_.timestamp >= offset)
+      .filter(_.timestamp > offset)
```

Comment:
Makes sense, we just have to remember to use a stateful filter when we use the query on our side 👍